app/models/content_page.rb

```diff
@@ -12,18 +12,22 @@ class ContentPage < ApplicationRecord
   validates :title, presence: true, uniqueness: true
   validates :markdown, presence: true
 
-  validates :position, presence: true, numericality: { only_integer: true }
-  validates :position, uniqueness: { scope: :parent_id }
+  validates :position, presence: true, numericality: { only_integer: true }, unless: -> { article? }
+  validates :position, uniqueness: { scope: :parent_id }, unless: -> { article? }
 
```

Comment:
These special cases are hints that something is wrong.   Articles will still need an order in the view.  It might be overridden, but anything is better than random